lib/rules/array-callback-return.js

```diff
@@ -188,19 +193,36 @@ module.exports = {
          * @returns {void}
          */
         function checkLastSegment(node) {
-            if (funcInfo.shouldCheck &&
-                funcInfo.codePath.currentSegments.some(isReachable)
-            ) {
-                context.report({
-                    node,
-                    loc: getLocation(node, context.getSourceCode()).loc.start,
-                    messageId: funcInfo.hasReturn
-                        ? "expectedAtEnd"
-                        : "expectedInside",
-                    data: {
-                        name: astUtils.getFunctionNameWithKind(funcInfo.node)
+            const method = astUtils.getStaticPropertyName(funcInfo.node.parent.callee);
+
+            if (funcInfo.shouldCheck) {
+                if (method === "forEach") {
+                    if (options.checkForEach &&
+                        !funcInfo.codePath.currentSegments.some(isReachable)
```

Comment:
I get what you mean, returning without a value as a way to stop the callback is ok, but returning any value, in any code path inside the callback is not allowed. The way it is now, it would allow a return value inside a code path, if another without a return statement existed. I will work on this